### PR TITLE
[feat] added support for stale-connection checks (maxPingOut, and pingInterval)

### PR DESCRIPTION
**Pull request by aricart**
Added support for stale connection checks.

### CODE REVIEW

File: nats-base-client/heartbeats.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  pendings: Promise<void>[] = [];
+
+  constructor(ph: PH, interval: number, maxOut: number) {
+    this.ph = ph;
+    this.interval = interval;
+    this.maxOut = maxOut;
+  }
+
+  // api to start the heartbeats, since this can be
+  // spuriously called from dial, insure we don't
```
Comment (kozlovic) [inline nats-base-client/heartbeats.ts:38]: `ensure`

File: nats-base-client/heartbeats.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  // leak timers
+  start() {
+    this.cancel();
+    this._schedule();
+  }
+
+  // api for canceling the heartbeats, if a string is
```
Comment (kozlovic) [inline nats-base-client/heartbeats.ts:45]: I don't understand `if a string is provided..` where?

Comment (aricart) [reply nats-base-client/heartbeats.ts:45]: stale comment.

File: nats-base-client/heartbeats.ts (hunk 1 above)
Comment (aricart) [reply nats-base-client/heartbeats.ts]: fixed

File: nats-base-client/heartbeats.ts (hunk 2 above)
Comment (aricart) [reply nats-base-client/heartbeats.ts]: fixed

**Pull request merged**